qubes/api/admin.py

```diff
@@ -1630,6 +1631,74 @@ async def vm_device_set_required(self, endpoint, untrusted_payload):
         await self.dest.devices[devclass].update_assignment(dev, mode)
         self.app.save()
 
+    @qubes.api.method(
+        "admin.vm.device.denied.List",
+        no_payload=True, scope="local", read=True)
+    async def vm_device_denied_list(self):
+        """
+        List all denied device interfaces for the VM.
+
+        Returns a newline-separated string.
+        """
+        self.enforce(not self.arg)
```

Comment:
This will result in `Got empty response from qubesd`, which indicates an internal error.  Please throw an exception that will be returned to the user of the Admin API.

Reply:
It's okay the way it is - the client part shouldn't put the argument there.